### ISSUE TITLE
Adam optimizer 

### ISSUE THREAD

**Issue by anshmishra010** [HarshCasper/NeoAlgo#7308]
## 🚀 Feature

Adam is an adaptive learning rate optimization algorithm that's been designed specifically for training deep neural networks.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

yes

## Motivation

in in python in ml folder

## Pitch

as a GSSoC'21
